In [1]:
import pandas as pd

In [5]:
df = pd.read_csv('prediction.csv')
df.head()

true_labels  predicted_labels  predcited_proba  \
0            0                 0         0.096054   
1            0                 0         0.415899   
2            0                 0         0.076353   
3            0                 0         0.348786   
4            0                 0         0.168456   

                      image_path  
0  test/NORMAL/IM-0001-0001.jpeg  
1  test/NORMAL/IM-0003-0001.jpeg  
2  test/NORMAL/IM-0005-0001.jpeg  
3  test/NORMAL/IM-0006-0001.jpeg  
4  test/NORMAL/IM-0007-0001.jpeg

In [9]:
eq = df[df.true_labels == df.predicted_labels]
neq = df[df.true_labels != df.predicted_labels]

In [ ]:
#eq[eq.true_labels == 0].sort_values('predcited_proba')
#eq[eq.true_labels == 1].sort_values('predcited_proba')
#neq[neq.true_labels == 0].sort_values('predcited_proba')
#neq[neq.true_labels == 1].sort_values('predcited_proba')

In [16]:
eq.sort_values('predcited_proba')

true_labels  predicted_labels  predcited_proba  \
162            0                 0         0.000258   
187            0                 0         0.000379   
185            0                 0         0.001561   
39             0                 0         0.001573   
206            0                 0         0.001758   
..           ...               ...              ...   
403            1                 1         0.999948   
488            1                 1         0.999963   
443            1                 1         0.999966   
359            1                 1         0.999970   
590            1                 1         0.999972   

                                     image_path  
162       test/NORMAL/NORMAL2-IM-0287-0001.jpeg  
187       test/NORMAL/NORMAL2-IM-0323-0001.jpeg  
185       test/NORMAL/NORMAL2-IM-0321-0001.jpeg  
39                test/NORMAL/IM-0063-0001.jpeg  
206       test/NORMAL/NORMAL2-IM-0346-0001.jpeg  
..                                          ...  
403  test/PNEUMONIA/person159_bacteria_747.jpeg  
488        test/PNEUMONIA/person1_virus_12.jpeg  
443   test/PNEUMONIA/person1651_virus_2855.jpeg  
359  test/PNEUMONIA/person141_bacteria_676.jpeg  
590   test/PNEUMONIA/person85_bacteria_422.jpeg  

[537 rows x 4 columns]

In [24]:
neq.sort_values('predcited_proba')

true_labels  predicted_labels  predcited_proba  \
317            1                 0         0.012161   
446            1                 0         0.014589   
599            1                 0         0.016087   
288            1                 0         0.021223   
282            1                 0         0.027803   
..           ...               ...              ...   
133            0                 1         0.992560   
160            0                 1         0.993893   
226            0                 1         0.997548   
144            0                 1         0.997621   
16             0                 1         0.999765   

                                     image_path  
317  test/PNEUMONIA/person127_bacteria_604.jpeg  
446   test/PNEUMONIA/person1655_virus_2861.jpeg  
599   test/PNEUMONIA/person88_bacteria_439.jpeg  
288  test/PNEUMONIA/person119_bacteria_568.jpeg  
282  test/PNEUMONIA/person117_bacteria_557.jpeg  
..                                          ...  
133       test/NORMAL/NORMAL2-IM-0232-0001.jpeg  
160       test/NORMAL/NORMAL2-IM-0285-0001.jpeg  
226       test/NORMAL/NORMAL2-IM-0373-0001.jpeg  
144       test/NORMAL/NORMAL2-IM-0256-0001.jpeg  
16                test/NORMAL/IM-0022-0001.jpeg  

[87 rows x 4 columns]

#### 2 примера правильно классифицированных с высокой вероятностью принадлежности к классу
1)1 - 1 - 0.999972 test/PNEUMONIA/person85_bacteria_422.jpeg

2)1 - 1 - 0.999970 test/PNEUMONIA/person141_bacteria_676.jpeg

?)0 -0 - 0.494304 test/NORMAL/NORMAL2-IM-0222-0001.jpeg

#### 2 примера правильно классифицированных с низкой вероятностью принадлежности к классу
1)0 - 0 - 0.000258 test/NORMAL/NORMAL2-IM-0287-0001.jpeg

2)0 - 0 - 0.000379 test/NORMAL/NORMAL2-IM-0323-0001.jpeg

?)1 -1 - 0.512877 test/PNEUMONIA/person1647_virus_2848.jpeg

#### 2 примера неправильно классифицированных с высокой вероятностью принадлежности к классу
1)0 - 1 - 0.999765 test/NORMAL/IM-0022-0001.jpeg

2)0 - 1 - 0.997621 test/NORMAL/NORMAL2-IM-0256-0001.jpeg

?)1 -0 - 0.482773 test/PNEUMONIA/person161_bacteria_762.jpeg

#### 2 примера неправильно классифицированных с низкой вероятностью принадлежности к классу
1)1 - 0 - 0.012161 test/PNEUMONIA/person127_bacteria_604.jpeg

2)1 - 0 - 0.014589 test/PNEUMONIA/person1655_virus_2861.jpeg

?)0 -1 - 0.501163 test/NORMAL/NORMAL2-IM-0171-0001.jpeg